pic-sure-auth-services/src/main/java/edu/harvard/hms/dbmi/avillach/auth/service/impl/RoleService.java

```diff
@@ -5,33 +5,91 @@
 import edu.harvard.hms.dbmi.avillach.auth.entity.User;
 import edu.harvard.hms.dbmi.avillach.auth.enums.SecurityRoles;
 import edu.harvard.hms.dbmi.avillach.auth.model.CustomUserDetails;
+import edu.harvard.hms.dbmi.avillach.auth.model.ras.RasDbgapPermission;
 import edu.harvard.hms.dbmi.avillach.auth.repository.PrivilegeRepository;
 import edu.harvard.hms.dbmi.avillach.auth.repository.RoleRepository;
+import edu.harvard.hms.dbmi.avillach.auth.utils.FenceMappingUtility;
+import jakarta.annotation.PostConstruct;
+import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
 import org.springframework.security.core.context.SecurityContext;
 import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import java.util.*;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
 @Service
 public class RoleService {
 
     private final Logger logger = LoggerFactory.getLogger(RoleService.class);
     private final RoleRepository roleRepository;
-
     private final PrivilegeRepository privilegeRepo;
     private final PrivilegeService privilegeService;
+    private final FenceMappingUtility fenceMappingUtility;
+    public static final String managed_open_access_role_name = "MANAGED_ROLE_OPEN_ACCESS";
+    private final Set<Role> publicAccessRoles = new HashSet<>();
 
     @Autowired
-    protected RoleService(RoleRepository roleRepository, PrivilegeRepository privilegeRepo, PrivilegeService privilegeService) {
+    public RoleService(RoleRepository roleRepository, PrivilegeRepository privilegeRepo, PrivilegeService privilegeService, FenceMappingUtility fenceMappingUtility) {
         this.roleRepository = roleRepository;
         this.privilegeRepo = privilegeRepo;
         this.privilegeService = privilegeService;
+        this.fenceMappingUtility = fenceMappingUtility;
+    }
+
+    @PostConstruct
+    public void init() {
+        // Log the public access roles as comma separated values for startup logs
+        String publicAccessRolesString = publicAccessRoles.stream().map(Role::getName).collect(Collectors.joining(", "));
+        logger.info("Public access roles: {}", publicAccessRolesString);
+        logger.info("RoleService initialized...");
+    }
+
+    @EventListener(ContextRefreshedEvent.class)
```

Comment:
We need to use the `ContextRefreshEvent` here because it executes this code after the ApplicationContext is initialized. This section on code depends on the FenceMappingUtility successfully completing its PostConstruct, which executes just after bean creation and before the ApplicationContext is completely initialized.